pkg/extensionAPI/errorHandlingExtension.go

```diff
@@ -119,6 +122,21 @@ func (e *JsExtension) AddInstance(context *context.ExtensionContext, version str
 	return e.extension.AddInstance(context, version, params), nil
 }
 
+func (e *JsExtension) SupportsListInstances(context *context.ExtensionContext, version string) bool {
+	if e.extension.FindInstances == nil {
+		return false
+	}
+	_, err := e.ListInstances(context, version)
+	if err == nil {
+		return true
+	}
+	apiErr := apiErrors.UnwrapAPIError(err)
+	if apiErr == nil {
+		return true
```

Comment:
The code tries `ListInstances` and fails with an error condition. Then the error is unwrapped to an API error and if that is `nil` everything is seemingly OK?

I get a glimpse of what this is supposed to do, but without the change log text I would immediately assume that this is an error.

Reply:
I refactored the code to make it more readable